src/lib/operators/join_mpsm/column_materializer_numa.hpp

```diff
@@ -123,33 +124,32 @@ class ColumnMaterializerNUMA {
 
     return std::make_shared<JobTask>(
         [this, &output, &null_rows_output, segment, chunk_id, alloc, numa_node_id] {
-          resolve_segment_type<T>(*segment, [&](auto& typed_segment) {
-            _materialize_segment(typed_segment, chunk_id, null_rows_output, (*output)[numa_node_id]);
-          });
+          if (const auto dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<T>>(segment)) {
+            _materialize_dictionary_segment(*dictionary_segment, chunk_id, null_rows_output, (*output)[numa_node_id]);
```

Comment:
What is the advantage of this change? To me it looks like we now need more code and a second method (instead of an overloaded one)

Reply:
Before, this code instantiated the lambda unnecessarily often.

This particular case is funny, because the let's-call-both-methods-_materialize_dictionary_segment-approach combined with a signature mismatch (probably due to a c/p mistake) led to the optimized dictionary variant never being called.